### DISCUSSION TITLE
How can i change kernel?

### DISCUSSION THREAD

**Discussion by ramide1** [SputnikRocket/ArchLinuxARM-rk3588-installer#3] [Q&A]
I want to use mainline kernel 6.7

**Comment by SputnikRocket** [comment]
Well I do not package mainline, you can probably get it from other sources, or compile it yourself. After you install it, you need to generate the initramfs, then run `grub-mkconfig -o /boot/grub/grub.cfg`. then you need to manually modify /boot/grub/grub.cfg, inserting `devicetree /path/to/dtb` before the `linux /vmlinu(z,x)<yourkernel>` line. sorry for not responding earlier.

I will put a proper tutorial for this on my todo.

**Comment by SputnikRocket** [comment]
@ramide1 I have tried booting mainline, but it does not appear to work when booted with UEFI for unknown reasons.

**Reply by ramide1** [reply]
yes, i have tried with armbian kernel but it does not work with uefi, it is strange

**Reply by ramide1** [reply]
hi, i have news, recently i tried 6.1 rockchip kernel and works fine with latest edk2 version on my orange pi 5

**Reply by SputnikRocket** [reply]
That's good news. I'm sorry that I have been inactive as of late.
